pkg/arvo/sur/sole.hoon

```diff
@@ -20,21 +21,22 @@
   $%  {$del p/@ud}                                      ::  delete one at
       {$ins p/@ud q/@c}                                 ::  insert at
       {$mor p/(list sole-edit)}                         ::  combination
-      {$nop ~}                                         ::  no-op
+      {$nop ~}                                          ::  no-op
       {$set p/sole-buffer}                              ::  discontinuity
   ==                                                    ::
 ++  sole-effect                                         ::  app to sole
-  $%  {$bel ~}                                         ::  beep
+  $%  {$bel ~}                                          ::  beep
       {$blk p/@ud q/@c}                                 ::  blink+match char at
-      {$clr ~}                                         ::  clear screen
+      {$clr ~}                                          ::  clear screen
       {$det sole-change}                                ::  edit command
       {$err p/@ud}                                      ::  error point
       {$klr p/styx}                                     ::  styled text line
       {$mor p/(list sole-effect)}                       ::  multiple effects
-      {$nex ~}                                         ::  save clear command
+      {$nex ~}                                          ::  save clear command
       {$pro sole-prompt}                                ::  set prompt
       {$sag p/path q/*}                                 ::  save to jamfile
       {$sav p/path q/@}                                 ::  save to file
+      {$tab p/(list {=term =type})}                     ::  tab-complete list
```

Comment:
I'm afraid I must object. It's not appropriate to send types in effects -- they might be massive and unsuitable for serialization. And I think this effect should be suitable for general purpose completions (chat might autocomplete audiences, for instance). It seems like it should be a list of something like `[term cord]`, `[cord cord]`, or `[cord tank]`. The latter can still be rendered at the correct width by drum.

Reply:
Ah, yes.  I figured for serialization you could just have the mark convert it to a tank.  But the point of making it general-purpose for things other than types is definitely correct.  I'll change it to `[cord tank]`.